File: core/src/ru/deadsoftware/cavedroid/game/mobs/MobsController.java
package ru.deadsoftware.cavedroid.game.mobs;

import com.badlogic.gdx.Gdx;
import ru.deadsoftware.cavedroid.game.GameScope;

import javax.inject.Inject;
import java.util.Iterator;
import java.util.LinkedList;

@GameScope
public class MobsController {

    public interface Callback {
        void run(Mob mob);
    }

    private static final String TAG = "MobsController";

    private final Player mPlayer;
    private final LinkedList<Mob> mMobs = new LinkedList<>();

    @Inject
    public MobsController() {
        mPlayer = new Player();
    }

    public Player getPlayer() {
        return mPlayer;
    }

    public void addMob(Class<? extends Mob> mobClass, float x, float y) {
        try {
            mMobs.add(mobClass.getConstructor(float.class, float.class).newInstance(x, y));
        } catch (Exception e) {
            Gdx.app.error(TAG, e.getMessage());
        }
    }

    public int getSize() {
        return mMobs.size();
    }

    public void forEach(Callback callback) {
        mMobs.forEach(callback::run);
    }

    public Iterator<Mob> getIterator() {
        return mMobs.iterator();
    }

}

File: core/src/ru/deadsoftware/cavedroid/game/mobs/MobsController.kt
package ru.deadsoftware.cavedroid.game.mobs

import ru.deadsoftware.cavedroid.game.GameItemsHolder
import ru.deadsoftware.cavedroid.game.GameScope
import ru.deadsoftware.cavedroid.game.mobs.player.Player
import ru.deadsoftware.cavedroid.game.ui.TooltipManager
import java.io.Serializable
import java.util.*
import javax.inject.Inject

@GameScope
class MobsController @Inject constructor(
    gameItemsHolder: GameItemsHolder,
    tooltipManager: TooltipManager,
) : Serializable {

    private val _mobs = LinkedList<Mob>()

    val player: Player =
        Player(gameItemsHolder, tooltipManager)

    val mobs: List<Mob>
        get() = _mobs

    fun addMob(mob: Mob) {
        _mobs.add(mob)
    }

    companion object {
        private const val TAG = "MobsController"
    }
}